consensus/src/epoch_manager.rs

```diff
@@ -663,7 +664,7 @@ impl<P: OnChainConfigProvider> EpochManager<P> {
 
     async fn init_payload_provider(
         &mut self,
-        epoch_state: &EpochState,
+        epoch_state: Arc<EpochState>,
```

Comment:
why?

Reply:
May not be super important for this particular PR. Basically, in the subsequent PR where the signed batch infos are also optimistically verified, we need to make sure the quroum store also sees the same view of epoch state.

Reply:
that makes sense, but probably should be in next pr to avoid confusion